src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -918,6 +918,23 @@ public void deleteLocalRepository(VcsRepositoryUrl repoUrl) {
         }
     }
 
+    /**
+     * delete the folder in the file system that contains all repositories for the given programming exercise
+     * @param programmingExercise contains the project key which is used as the folder name
+     */
+    public void deleteLocalProgrammingExerciseReposFolder(ProgrammingExercise programmingExercise) {
+        var folderPath = Paths.get(repoClonePath, programmingExercise.getProjectKey());
+        try {
+            FileUtils.deleteDirectory(folderPath.toFile());
+        }
+        catch (IOException ex) {
+            log.error("Exception during deleteLocalProgrammingExerciseReposFolder " + ex, ex);
```

Comment:
Not sure if it's important but should it be `ex.getMessage()` like the changes written below?